colibris/views.py

```diff
@@ -179,10 +199,28 @@ def get_schema(self, *args, **kwargs):
 
         return json_payload
 
+    async def get_validated_query(self, schema=None):
+        if schema is None:
+            schema = self.get_query_schema_class()
+
+        assert schema is not None, 'The attribute "query_schema_class" is required for {}'.format(self)
+
+        json_payload = await self.get_request_query()
+
+        try:
+            data = schema.load(json_payload)
+        except ValidationError as err:
+            raise api.SchemaError(details=err.messages)
+
+        return data
+
+    async def get_request_query(self):
+        return self.request.query
+
 
 class ModelView(APIView):
     model = Model
-    schema_class = ModelSchema
+    body_schema_class = ModelSchema
```

Comment:
We should probably set these to `None` and complain if they are unset. Having them defaulting to base `Model` and `ModelSchema` wouldn't help anyone, I think.

Reply:
Could be an option. The problem is that `_GenericMixinMeta` is used when the classes are imported, so, if the class `ModelView` has `body_schema_class` to `None`,   `_GenericMixinMeta` will throw an error. Should we go with default as `None` and check if the class is a base class and complain only on "final" classes?

Reply:
I definitely favor the `None` approach but am not sure if we should do any assertions. Like I said before, we don't usually do this kind of validations throughout Colibris and having them here seems kind of inconsistent to me.

Long story short: if we went with duck typing from the beginning, let's carry on this way.